packages/react-reconciler/src/ReactFiberHooks.new.js

```diff
@@ -913,7 +913,16 @@ function readFromUnsubcribedMutableSource<Source, Snapshot>(
   }
 
   if (isSafeToReadFromSource) {
-    return getSnapshot(source._source);
+    const snapshot = getSnapshot(source._source);
+    if (__DEV__) {
+      if (typeof snapshot === 'function') {
+        console.warn(
```

Comment:
Should it be console.error? Seems very error like if it might give wrong output and we might make it an invariant.